workspaces/orchestrator/plugins/orchestrator-backend/src/service/router.ts

```diff
@@ -688,22 +800,23 @@ function setupInternalRoutes(
         message: `Received request to '${endpoint}' endpoint`,
       });
 
-      const decision = await authorize(
-        req,
-        orchestratorWorkflowInstancesReadPermission,
-        permissions,
-        httpAuth,
-      );
-      if (decision.result === AuthorizeResult.DENY) {
-        manageDenyAuthorization(endpointName, endpoint, req);
+      try {
+        // TODO: use pagination only if the generic orchestratorWorkflowPermission is in place
```

Comment:
I think there's an issue here
Let's say there are 100 instances and user is authorized to view 50
amongst the first 20, some of them he's not allowed to view
He'll see 10 instances instead of 20
Is there a way to get the workflow ids he's allowed to view and add them to the graphql filter?

Reply:
At this moment I am not aware of a way how to get this list.

This flow can be another way how to implement https://issues.redhat.com/browse/FLPATH-1916.
If we can get such list of user's permissions, it will be much easier than storing user info in an instance.

Reply:
RBAC doesn't have API for getting list of permissions?

Reply:
https://github.com/backstage/community-plugins/blob/main/workspaces/rbac/plugins/rbac-backend/docs/apis.md#permission

Reply:
The RBAC UI displays list of permissions of selected user so there must be a way to get it

Reply:
OK I took a deeper look and it indeed seems like there's no way to achieve this
I don't see how we can implement the pagination and the RBAC restriction of list of instances together in a clean way
Maybe better list all of the instances and block getInstanceById and  rerun in the instance page